facades/PC/src/main/java/org/terasology/engine/TerasologyCommand.java

```diff
@@ -0,0 +1,417 @@
+import org.terasology.engine.subsystem.lwjgl.LwjglInput;
+import org.terasology.engine.subsystem.lwjgl.LwjglTimer;
+import org.terasology.engine.subsystem.openvr.OpenVRInput;
+import org.terasology.engine.subsystem.rpc.DiscordRPCSubSystem;
+import org.terasology.game.GameManifest;
+import org.terasology.network.NetworkMode;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameInfo;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameProvider;
+import org.terasology.splash.SplashScreen;
+import org.terasology.splash.SplashScreenBuilder;
+import org.terasology.splash.overlay.AnimatedBoxRowOverlay;
+import org.terasology.splash.overlay.ImageOverlay;
+import org.terasology.splash.overlay.RectOverlay;
+import org.terasology.splash.overlay.TextOverlay;
+import org.terasology.splash.overlay.TriggerImageOverlay;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static com.google.common.base.Verify.verifyNotNull;
+
+/**
+ * Class providing the main() method for launching Terasology as a PC app.
+ * <br><br>
+ * Through the following launch arguments default locations to store logs and
+ * game saves can be overridden, by using the current directory or a specified
+ * one as the home directory. Furthermore, Terasology can be launched headless,
+ * to save resources while acting as a server or to run in an environment with
+ * no graphics, audio or input support. Additional arguments are available to
+ * reload the latest game on startup and to disable crash reporting.
+ * <br><br>
+ * Available launch arguments:
+ * <br><br>
+ * <table summary="Launch arguments">
+ * <tbody>
+ * <tr><td>-homedir</td><td>Use the current directory as the home directory.</td></tr>
+ * <tr><td>-homedir=path</td><td>Use the specified path as the home directory.</td></tr>
+ * <tr><td>-headless</td><td>Start headless.</td></tr>
+ * <tr><td>-loadlastgame</td><td>Load the latest game on startup.</td></tr>
+ * <tr><td>-createlastgame</td><td>Recreates the world of the latest game with a new save file on startup.</td></tr>
+ * <tr><td>-noSaveGames</td><td>Disable writing of save games.</td></tr>
+ * <tr><td>-noCrashReport</td><td>Disable crash reporting.</td></tr>
+ * <tr><td>-noSound</td><td>Disable sound.</td></tr>
+ * <tr><td>-noSplash</td><td>Disable splash screen.</td></tr>
+ * <tr><td>-serverPort=xxxxx</td><td>Change the server port.</td></tr>
+ * </tbody>
+ * </table>
+ * <br><br>
+ * When used via command line an usage help and some examples can be obtained via:
+ * <br><br>
+ * terasology -help    or    terasology /?
+ * <br><br>
+ */
+
+@CommandLine.Command(name = "terasology")
+public class TerasologyCommand implements Callable<Integer> {
```

Comment:
Does the `Callabale` move this to a different thread? Will this cause even more issues with porting to LWJGL 3 and MacOS (see #3969 )?

Reply:
I don't *think* so. I think picocli just uses Callable<Integer> as an interface that's easy to define for use with its `.execute(it)` method, so it can run it, handle exceptions, and get a process return code at the end. But it'll be a good thing to double-check.

Reply:
I confirmed this. `currentThread()` returns the same value inside the `call` method as it does in the static `main`.